hs_tracking/models.py

```diff
@@ -0,0 +1,147 @@
+from json import loads, dumps
+from datetime import datetime, timedelta
+
+from django.db import models
+from django.core import signing
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+from theme.models import UserProfile
+
+
+SESSION_TIMEOUT = 60 * 15
+PROFILE_FIELDS = ["title", "user_type", "subject_areas", "public", "state", "country"]
+USER_FIELDS = ["username", "email", "first_name", "last_name"]
+VISITOR_FIELDS = ["id"] + USER_FIELDS + PROFILE_FIELDS
+if set(PROFILE_FIELDS) & set(USER_FIELDS):
+    raise ImproperlyConfigured("hs_tracking PROFILE_FIELDS and USER_FIELDS must not contain overlapping field names")
+
+
+class SessionManager(models.Manager):
+    def for_request(self, request):
+        signed_id = request.session.get('hs_tracking_id')
+        if signed_id:
+            tracking_id = signing.loads(signed_id)
+            cut_off = datetime.now() - timedelta(seconds=SESSION_TIMEOUT)
+            session = None
+            try:
+                session = Session.objects.filter(begin__gte=cut_off).get(id=tracking_id['id'])
```

Comment:
You're right. I'll pivot to apply the session timeout not to the start of the session but to the last variable tracked in that session. That was actually my intention and I overlooked this during implementation.